corehq/apps/sms/filters.py

```diff
@@ -38,6 +38,18 @@ def options(self):
         return options_var
 
 
+class ErrorCodeFilter(BaseMultipleOptionFilter):
+    label = ugettext_noop('Error')
+    default_text = ugettext_noop('Select Error...')
+    slug = 'error_code'
+
+    options = [
+        (code, message.split(".")[0])   # shorten multi-sentence messages
```

Comment:
thought about this a bit. Just noting down thoughts, if any feel useful/worth thinking about.

The table would still have the complete error message so this is probably okay where the message is just a part in the select box. I see 2 messages that are really long but all others are okay so not sure if this is really necessary.
How annoying is it to see a long message in that select box? Does select2 adjust for the length in a nice way?

I was also thinking if we just keep the code or a humanized version of the code as the value here. Most of them are pretty straight forward.

Also, would be even nicer, is if we could add a tooltip on hover over in the table, for status column, where when someone hovers over the value in column "status", it reveals the complete message. And we can just keep the code for the search and the value.

Reply:
The longest error messages get cut off in an ugly way in the select2 box.

What would the value be of making error code visible in any way? I agree the error codes are fairly human-readable, but it's better for end users to see the error message that was written with them in mind. Error codes also aren't translated.

Reply:
In general hovering is better to avoid, since the user then has to pause on every item individually, plus it doesn't translate well to mobile.

Reply:
> The longest error messages get cut off in an ugly way in the select2 box.

Have you thought of shortening the long ones? I believe we store the error code so it should be possible to just make the messages short.

> What would the value be of making error code visible in any way?
In general hovering is better to avoid, since the user then has to pause on every item individually

I was thinking that these codes don't really change much, so if someone uses that report for a few times they would know about the codes. Depends a bit of the kind of users who would be using this report.

> Error codes also aren't translated.

Oh yes, good point. This would then become new translations.

> In general hovering is better to avoid, since the user then has to pause on every item individually, plus it doesn't translate well to mobile.

Right. Definitely not a good UX. 

Thanks for talking this through and sharing the challenges.

Reply:
> Have you thought of shortening the long ones?

I'm not quite following (or maybe didn't explain clearly), that's what this code is doing. When the messages are limited to one sentence, they're short enough to fit in the search box.

Reply:
Oh, I meant changing the original message so its small enough that we don't need to split it here.

Reply:
Going to leave this as-is. The longer error messages have info on how to deal with the error, so I think it's worth keeping but doesn't need to clutter up the search box: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/sms/models.py#L990

Reply:
Cool, thanks for hearing me out.